protocols/gossipsub/src/behaviour/tests.rs

```diff
@@ -22,57 +22,307 @@
 
 #[cfg(test)]
 mod tests {
```

Comment:
How about not wrapping the entire content of the file in the `mod tests {}`, but instead move the `#[cfg(test)]` to `lib.rs` where the `tests.rs` module is introduced?